Apps/VisionOS/VisionOS/MainScreen.swift

```diff
@@ -1,11 +1,23 @@
 import SwiftUI
 
 struct MainScreen: View {
+    @ObservedObject var state: AppState = .shared
     var body: some View {
         VStack {
             Text("Hello, world!")
         }
         .padding()
+        .environment(
+            \.openURL,
+            OpenURLAction { url in
+                guard let link = InlineNavigationLink(fromUrl: url) else {
+                    return .systemAction
+                }
+                state.navigationPath = [link]
```

Comment:
Why do we need to save this navigation link into the AppState? 

I was expecting the app to navigate to a screen and that's it.

Reply:
this will cause the next app run to start on the screen it was terminated at instead of the very initial screen as the first app run